react/src/components/Table/Table.tsx

```diff
@@ -71,34 +72,103 @@ const resolveSex = (sexPhenotype: string) => {
     } else return 'Unknown';
 };
 
-/* flatten data and compute values as needed (note that column display formatting function should not alter values for ease of export) */
-const prepareData = (queryResult: VariantQueryDataResult[]): ResultTableColumns[] => {
-    return queryResult.flatMap(d => {
-        if (d.variant.callsets.length) {
-            //one row per individual per callset
-            return d.variant.callsets
-                .filter(cs => cs.individualId === d.individual.individualId)
-                .map(cs =>
-                    addAdditionalFieldsAndFormatNulls({
-                        ...cs.info,
-                        ...flattenBaseResults(d),
-                    })
-                );
+/* 1, Sort queryResult in ascending order according to variant's chromosome, ref, alt, start,end. 
+2, flatten data and compute values as needed 
+(note that column display formatting function should not alter values for ease of export) */
+const prepareData = (queryResult: VariantQueryDataResult[]): [ResultTableColumns[], number[]] => {
+    const sortedQueryResult = [...queryResult].sort(function (a, b) {
+        if (a.variant.referenceName < b.variant.referenceName) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref < b.variant.ref
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt < b.variant.alt
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt === b.variant.alt &&
+            a.variant.start < b.variant.start
+        ) {
+            return -1;
+        } else if (
+            a.variant.referenceName === b.variant.referenceName &&
+            a.variant.ref === b.variant.ref &&
+            a.variant.alt === b.variant.alt &&
+            a.variant.start === b.variant.start &&
+            a.variant.end < b.variant.end
+        ) {
+            return -1;
         } else {
-            return addAdditionalFieldsAndFormatNulls(flattenBaseResults(d));
+            return 1;
         }
     });
+
+    var result: Array<ResultTableColumns> = [];
+    var uniqueVariantIndices: Array<number> = []; // contains indices of first encountered rows that represent unique variants.
+    const n = sortedQueryResult.length;
+    var currVariant = { chromosome: 'temp', ref: 'temp', alt: 'temp', start: 0, end: 0 };
```

Comment:
```suggestion
   type Variant = Pick<VariantResponseFields, 'referenceName' | 'alt' | 'ref' | 'start' | 'end'>;
    var currVariant: Variant  = {} as Variant;
```

Note: You should move the `type Variant` to top of file where all the other types are declared.